src/cron-schedule.ts

```diff
@@ -0,0 +1,206 @@
+      public _bind() {
+        return new LiteralSchedule(this.expressionString);
+      }
+    };
+  }
+
+  /**
+   * Create a cron schedule which runs every day at midnight
+   */
+  public static daily(): CronSchedule {
+    const minute = 0;
+    const hour = 0;
+    const day = '*';
+    const month = '*';
+    const weekDay = '?';
+
+    return new class extends CronSchedule {
+      public readonly expressionString: string = `${minute} ${hour} ${day} ${month} ${weekDay}`;
+      public _bind() {
+        return new LiteralSchedule(this.expressionString);
+      }
+    };
+  }
+
+  /**
+   * Create a cron schedule which runs every week on Sunday
+   */
+  public static weekly(): CronSchedule {
+    const minute = 0;
+    const hour = 0;
+    const day = '?';
+    const month = '*';
+    const weekDay = 0;
+
+    return new class extends CronSchedule {
+      public readonly expressionString: string = `${minute} ${hour} ${day} ${month} ${weekDay}`;
+      public _bind() {
+        return new LiteralSchedule(this.expressionString);
+      }
+    };
+  }
+
+  /**
+   * Create a cron schedule which runs first day of every month
+   */
+  public static monthly(): CronSchedule {
+    const minute = 0;
+    const hour = 0;
+    const day = 1;
+    const month = '*';
+    const weekDay = '?';
+
+    return new class extends CronSchedule {
+      public readonly expressionString: string = `${minute} ${hour} ${day} ${month} ${weekDay}`;
+      public _bind() {
+        return new LiteralSchedule(this.expressionString);
+      }
+    };
+  }
+
+  /**
+   * Create a cron schedule which runs first day of January every year
+   */
+  public static annually(): CronSchedule {
+    const minute = 0;
+    const hour = 0;
+    const day = 1;
+    const month = 1;
+    const weekDay = '?';
+
+    return new class extends CronSchedule {
+      public readonly expressionString: string = `${minute} ${hour} ${day} ${month} ${weekDay}`;
+      public _bind() {
+        return new LiteralSchedule(this.expressionString);
+      }
+    };
+  }
+
+  /**
+   * Create a custom cron schedule from a set of cron fields
+   */
+  public static custom(options: CronOptions): CronSchedule {
+    if (options.weekDay !== undefined && options.day !== undefined) {
+      throw new Error('Cannot supply both \'day\' and \'weekDay\', use at most one');
+    }
+
+    const minute = fallback(options.minute, '*');
+    const hour = fallback(options.hour, '*');
+    const month = fallback(options.month, '*');
+
+
+    // Weekday defaults to '?' if not supplied. If it is supplied, day must become '?'
+    const day = fallback(options.day, options.weekDay !== undefined ? '?' : '*');
+    const weekDay = fallback(options.weekDay, '?');
+
+    return new class extends CronSchedule {
+      public readonly expressionString: string = `${minute} ${hour} ${day} ${month} ${weekDay}`;
+      public _bind() {
+        return new LiteralSchedule(this.expressionString);
+      }
```

Comment:
What is this? Who is supposed to call it? I think these are only relevant for the AWS CDK.

Intuitively the `CronSchedule` class should just a single readonly property and a private constructor.

Reply:
Ok, will make the change. I kept it around thinking we need to have it a bit similar to CDK. :)